src/odemis/util/focus.py

```diff
@@ -120,6 +120,13 @@ def MeasureSpotsFocus(image):
     image (model.DataArray): Optical image
     returns (float): The focus level of the image (higher is better)
     """
+    # OpenCV only supports int of 8 & 16 bits and float32/64 images. So if we get anything else,
+    # we first convert to float64, to make sure it's compatible.
+    if not (
+            (numpy.issubdtype(image.dtype, numpy.integer) and image.dtype.itemsize <= 2)
+            or image.dtype in (numpy.float32, numpy.float64)
+        ):
+         image = image.astype(numpy.float64)
```

Comment:
I think this might also be a problem in `MeasureOpticalFocus`. If you agree could you add the check there as well?

Reply:
Indeed.
Or maybe we should switch to scipy? 
See:
scipy.ndimage.sobel()
scipy.ndimage.laplace()

At *first* look they seem equivalent, and available already on Ubuntu 20.04 (Scipy 1.3.3):
https://docs.scipy.org/doc/scipy-1.3.1/reference/generated/scipy.ndimage.laplace.html

Reply:
As we want to have it merged soon, I've just extended the change as-is to MeasureOpticalFocus(). We can later think of switching to scipy.

Reply:
I vaguely remember that the scipy implementation was a lot slower. But this was 5 years ago, so maybe it has improved. 

I agree with leaving the change as is and thinking later of switching to scipy